src/actions/hubspot/hubspot.ts

```diff
@@ -0,0 +1,257 @@
+import * as hubspot from "@hubspot/api-client"
+import * as semver from "semver"
+import * as util from "util"
+import * as winston from "winston"
+import * as Hub from "../../hub"
+import { RequiredField } from "../../hub"
+import { HubspotActionError } from "./hubspot_error"
+
+export enum HubspotTags {
+  ContactId = "hubspot_contact_id",
+  CompanyId = "hubspot_company_id",
+}
+
+export enum HubspotCalls {
+  Contact = "contact",
+  Company = "company",
+}
+
+interface DefaultHubspotConstructorProps {
+  name: string
+  label: string
+  description: string
+  call: HubspotCalls
+  tag: HubspotTags
+}
+
+async function delay(ms: number) {
+  // tslint continually complains about this function, not sure why
+  // tslint:disable-next-line
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+const HUBSPOT_BATCH_UPDATE_DEFAULT_LIMIT = 10
+const HUBSPOT_BATCH_UPDATE_ITERATION_DELAY_MS = 500
+
+export class HubspotAction extends Hub.Action {
+  name: string
+  label: string
+  description: string
+  call: HubspotCalls
+  tag: HubspotTags
+  requiredFields: RequiredField[]
+
+  iconName = "hubspot/hubspot.png"
+  params = [
+    {
+      description: "An api key for Hubspot.",
+      label: "Hubspot API Key",
+      name: "hubspot_api_key",
+      required: true,
+      sensitive: true,
+    },
+    {
+      description:
+        "The number of objects to batch update per call (defaulted to 10)",
+      label: "Batch Update Size",
+      name: "hubspot_batch_update_size",
+      required: false,
+      sensitive: false,
+    },
+  ]
+  supportedActionTypes = [Hub.ActionType.Query]
+  usesStreaming = true
+  supportedFormattings = [Hub.ActionFormatting.Unformatted]
+  supportedVisualizationFormattings = [
+    Hub.ActionVisualizationFormatting.Noapply,
+  ]
+  executeInOwnProcess = true
```

Comment:
I am glad you did this already. Out of curiosity what is the biggest dataset you have tried streaming through this action and how big do you expect a "normal" request to be?

Reply:
@phillipperalez depends entirely on the size of the company, for us we're passing in a few thousand at a time, upwards of maybe 5k rows from Looker.